asmcomp/clambda.ml

```diff
@@ -62,8 +70,8 @@ and ulambda =
 and ufunction = {
   label  : function_label;
   arity  : int;
-  params : Ident.t list;
-  body   : ulambda;
+  params : (Ident.t * value_kind) list;
+  body   : (ulambda * value_kind);
```

Comment:
Why do you put the return type here rather than in a 'return_type' field ?

Reply:
This seemed coherent with the handling of parameters.  The value_kind describe the type of values returned by the ulambda in body.  This "forces" to think about updating the result type if body is changed for whatever reason.

Reply:
In #1029, you had to be careful for:

https://github.com/ocaml/ocaml/pull/1029/files#diff-704f66c0fa0fc9339230b39ce7d90919R464

While here, I have:

https://github.com/ocaml/ocaml/pull/1322/files#diff-704f66c0fa0fc9339230b39ce7d90919R468

("clearly", since the "body" of the result is the one from the inner function, the "value_kind" should also be taken from that function)